### ISSUE TITLE
Define new resource type azuredevops_git_repository

### ISSUE THREAD

**Issue by KeithJRome** [microsoft/terraform-provider-azuredevops#93]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Define a new resource type for creating git repositories in a devops project:
- Stubs for Create, Update, Read, Delete
- Simple schema, based on https://github.com/microsoft/azure-devops-go-api/blob/dev/azuredevops/git/models.go#L1875
  - [input, required] project
  - [input, required] name
  - [input, optional] default_branch
  - etc...

### New or Affected Resource(s)

* *new* azuredevops_git_repository resource

### Potential Terraform Configuration

```hcl
resource "azuredevops_project" "main_project" {
  project_name       = "Test Project"
  description        = "Test Project Description"
  visibility         = "private"
  enable_tfvc        = False
  work_item_template = "Agile"
}

resource "azuredevops_git_repository" "main" {
  project        = azuredevops_project.main_project.id
  name           = "main-repo"
  default_branch = "master"
}

resource "azuredevops_git_repository" "secondary" {
  project        = azuredevops_project.main_project.id
  name           = "secondary-repo"
  default_branch = "master"
}
```

### References

* Story spawned from design task #73

**Comment by tmeckel** [comment]
I think it's important to have a Terraform data source as well for git repositories. Especially when it comes to manage security and permissions with this Terraform provider.

The data source should export at least a property `id`, so that the data source could be used in other resources.

## Potential Terraform Configuration

```terraform
data "azuredevops_git_repository" "main" {
  project        = azuredevops_project.main_project.id
  name          = "secondary-repo"
}
```

**Comment by KeithJRome** [comment]
Thanks for your contributions @tmeckel 

We've discussed this internally and agree that there should be a data source peer for this resource, as well as in other cases that we hadn't previously considered, to increase flexibility of this provider. I'll split off the data source work into a separate story so that it can live on the backlog and get prioritized. You're of course free to implement and submit a PR for it too (you don't need to wait on our team).

**Comment by tmeckel** [comment]
Hi @KeithJRome added this as a new issue.

https://github.com/microsoft/terraform-provider-azuredevops/issues/106

Currently we are in a planning phase for automatic deployment of DevOps artifacts like projects, Git repositories and the like in a cloud project at a German bank. We could like to get touch with the team here how to discuss how to provide support for implementing functionality. I copy @tiwood for reference here.

**Comment by nmiodice** [comment]
@tmeckel thank you for reaching out. I want to make our internal roadmap more visible so I've posted a PR here: https://github.com/microsoft/terraform-provider-azuredevops/pull/111. Our short term priorities are stemming from this road-map so our efforts will be focused here (in the short term) but we always appreciate open source contributions.

Have you come across our [contributing guidelines](https://github.com/microsoft/terraform-provider-azuredevops/blob/master/docs/contributing.md)?

**Comment by tmeckel** [comment]
> 
> 
> @tmeckel thank you for reaching out. I want to make our internal roadmap more visible so I've posted a PR here: #111. Our short term priorities are stemming from this road-map so our efforts will be focused here (in the short term) but we always appreciate open source contributions.
> 
My colleague (@tiwood) and I would like to support the project. However, it would make sense if we could find a way to avoid redundant work with other people here. Like assigning issues to one of us so that it'll be clear that the issue is currently worked on and how's responsible. I don't know if you'd like to run the project that way.

> Have you come across our [contributing guidelines](https://github.com/microsoft/terraform-provider-azuredevops/blob/master/docs/contributing.md)?

My colleague and I have already worked on various Open Source projects and are very familiar with the conventions. 

Personally I commited, amongst others, to SharePoint PnP, DSC modules and the OpenSSH Windows Project and thus already signed the CLA and I'm familiar how Microsoft runs open source projects here on GitHub. And I commited the [VirtualBox VM builder](https://github.com/hashicorp/packer/pull/7780) for Hashicorp Packer so I know how those guys run their projects as well.

@tiwood commited the Azure Terraform data source [azurerm_resources](https://github.com/terraform-providers/terraform-provider-azurerm/pull/3529) aside many other improvements for the Terraform and Terragrunt projects; e.g. (https://github.com/terraform-providers/terraform-provider-azurerm/pulls?utf8=%E2%9C%93&q=is%3Apr+author%3Atiwood)

**Comment by nmiodice** [comment]
@tmeckel @tiwood I'd love to continue this discussion in a more chat-friendly forum. I've gone ahead and created a gitter for us to chat: https://gitter.im/terraform-provider-azuredevops/community#

I'd love to hear more about your use cases to see what kind of alignment there is between our work. Can we move the discussion to https://gitter.im/terraform-provider-azuredevops/community#?
